packages/core/src/utils.ts

```diff
@@ -726,7 +726,12 @@ for (let n = 0; n < 256; n++) {
  * @returns The resulting hex string.
  */
 export function arrayBufferToHex(arrayBuffer: ArrayBuffer): string {
-  const bytes = new Uint8Array(arrayBuffer);
+  // We want to allow both views and raw ArrayBuffers, so we conditionally "unwrap" the underlying ArrayBuffer
+  // If `arrayBuffer` was a view
+  // Otherwise, each of the views "elements" (ie. a uint32) will be truncated to a uint8
+  // And you end up with an incorrect shortened result of the same number of bytes as there were elements in the original TypedArray
+  const buffer = ArrayBuffer.isView(arrayBuffer) ? arrayBuffer.buffer : arrayBuffer;
+  const bytes = new Uint8Array(buffer);
```

Comment:
Could this logic be extracted into a utility function called in both exported functions?